auth0/resource_auth0_organization_member.go

```diff
@@ -0,0 +1,143 @@
+package auth0
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/auth0/go-auth0/management"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func newOrganizationMember() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: createOrganizationMember,
+		ReadContext:   readOrganizationMember,
+		UpdateContext: updateOrganizationMember,
+		DeleteContext: deleteOrganizationMember,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"organization_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The ID of the organization to assign member to",
+			},
+			"user_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "ID of the user to add as organization member",
+			},
+			"roles": {
+				Type:        schema.TypeSet,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "Role ID(s) to assign to member",
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func createOrganizationMember(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	userID := String(d, "user_id")
+	orgID := String(d, "organization_id")
+
+	d.SetId(resource.UniqueId())
+
+	api := m.(*management.Management)
+	if err := api.Organization.AddMembers(*orgID, []string{*userID}); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := assignRoles(d, m); err != nil {
+		return diag.FromErr(fmt.Errorf("failed to assign roles to organization member. %w", err))
+	}
+
+	return readOrganizationMember(ctx, d, m)
+}
+
+func assignRoles(d *schema.ResourceData, m interface{}) error {
```

Comment:
That's a good suggestion. However, the `updateOrganizationMember` function which is also a caller of `assignRoles`, does not have those values readily available. So it would necessitate duplicate calculations of changing roles.As-is, it is a bit of an amalgamation of functionality in a single function, but I think it makes just enough sense in the context of this resource.